python/lsst/daf/butler/core/datasets.py

```diff
@@ -50,15 +51,38 @@ class DatasetType(object):
     dataUnits : `iterable` of `str`
         `DataUnit` names that defines the `DatasetRef`\ s corresponding to
         this `DatasetType`.  The input iterable is copied into a `frozenset`.
-    storageClass : `str`
-        Name of a `StorageClass` that defines how this `DatasetType`
+    storageClass : `StorageClass`
+        Instance of a `StorageClass` that defines how this `DatasetType`
         is persisted.
     """
 
     __slots__ = ("_name", "_dataUnits", "_storageClass")
     __eq__ = slotValuesAreEqual
     __hash__ = slotValuesToHash
 
+    def __str__(self):
+        return "DatasetType({}, {}, {})".format(self.name, self.storageClass.name, self.dataUnits)
+
+    @staticmethod
+    def nameWithComponent(datasetTypeName, componentName):
```

Comment:
@pschella you might have an opinion on `nameWithComponent`, `component` and `componentTypeName` in `DatasetType`. They contain the logic for "calexp.wcs" naming convention so I wanted it to be in the one place in this class.